framework/include/utils/ContactLineSearch.h

```diff
@@ -0,0 +1,116 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef CONTACTLINESEARCH_H
+#define CONTACTLINESEARCH_H
+
+#include "libmesh/petsc_nonlinear_solver.h"
+
+#ifdef LIBMESH_HAVE_PETSC
```

Comment:
Well this isn't going to work. It's hard for me to picture what MOOSE would look like without PETSc these days but conditionally defining an entire object, then including it **always** inside of another simply won't work. If you want to do this properly, you should define this object in any case but print out a useful runtime error when PETSc isn't configured.

Now if you think not worrying about _not_ having PETSc is pedantic, I agree with you and we'll have to address this with the rest of the team.